### DISCUSSION TITLE
CHIP-8 Emulator

### DISCUSSION THREAD

**Discussion by stewlab** [shorepine/tulipcc#289] [Show and tell]
I have been working on a CHIP-8 emulator / interpreter for Tulip CC, and recently got external program loading and display working.  I will create a public repository for the code soon, and will also share it in Tulip World.

CHIP-8 is essentially the original fantasy console, in a similar category as [TIC-80](https://en.wikipedia.org/wiki/TIC-80) and [PICO-8](https://en.wikipedia.org/wiki/PICO-8).

CHIP-8 Info:
https://en.wikipedia.org/wiki/CHIP-8

Here is a screenshot of the app running an IBM Logo CHIP-8 program via Tulip Desktop: 
![Screenshot from 2024-08-03 02-23-00](https://github.com/user-attachments/assets/59d586f3-2814-41cf-9655-349dd848e08d)

Here is a video of another CHIP-8 demo running in the same environment:
https://github.com/user-attachments/assets/927c5bfc-7b66-49f0-8233-e3cd8dd5e23d

**Comment by bwhitman** [comment]
Wow!!! Super cool!

**Comment by stewlab** [comment]
Here is the initial release:
https://github.com/stewlab/tulip8

It seems to work on Tulip Desktop and a Tulip Creative Computer device, but not quite on T-Deck just yet. Also, performance is pretty bad, so I am still working on optimization.
